sort_fastq/sorting_fastq.py

```diff
@@ -0,0 +1,88 @@
+import re
+import os
+
+
+def process_batch(batch):
+    batch[0] = re.sub(r'\n|@', '', batch[0])  # удаляю перенос строки и символ @ из идентификатора
+    for i in range(1, len(batch)):
+        batch[i] = re.sub(r'\n', '', batch[i])  # у всех остальных удаляю перенос строки
+
+
+class Read:
+    def __init__(self, read_id, read_sequence, comment, quality):
+        self.read_id = read_id
+        self.read_sequence = read_sequence
+        self.comment = comment
+        self.quality = quality
+
+    def gc(self):
+        read_sequence = self.read_sequence
+        gc_content = sum(map(lambda x: x == "C" or x == "G", read_sequence))/len(read_sequence)
+        return gc_content
+
+    def mean_quality(self):
+        quality = self.quality
+        mean = sum(map(lambda x: ord(x) - 33, quality)) / len(quality)
+        return mean
+
+    def __len__(self):
+        read_sequence = self.read_sequence
+        return len(read_sequence)
+
+
+class FASTQFile:
+    def __init__(self, fastq_records, file_path):  # file_path - путь до исходного fastq файла
+        self.fastq_records = fastq_records
+        self.file_path = file_path
+
+    def sort_reads(self):
+        stat_per_read = {}   # словарь (статистика рида): объект рида
+        for fastq in self.fastq_records:
+            stat_per_read[(fastq.mean_quality(), len(fastq), fastq.gc(), fastq.read_id)] = fastq
+        stat = [key for key in stat_per_read]   # положим все статистики ридов в виде кортежей в список
+        stat = sorted(stat)     # сортируем кортежи внутри списка
+        for k, i in enumerate(stat):    # вернемся к объектам из сортированных статистик
+            fastq = stat_per_read[i]
+            self.fastq_records[k] = fastq   # изменим исходный атрибут
```

Comment:
Я имел ввиду вот такую запись
```suggestion
        self.fastq_records.sort(key=lambda read: (read.mean_quality(), len(read), read.gc(), read.read_id))
```